sw/device/lib/dif/dif_i2c.h

```diff
@@ -567,6 +567,18 @@ dif_result_t dif_i2c_get_fifo_levels(const dif_i2c_t *i2c,
 OT_WARN_UNUSED_RESULT
 dif_result_t dif_i2c_read_byte(const dif_i2c_t *i2c, uint8_t *byte);
 
+/**
+ * Reads off a chunk of bytes from the RX FIFO.
+ *
+ * @param i2c An I2C handle.
+ * @param[out] size The size of the buffer.
+ * @param[out] buffer A buffer to receive the bytes read.
+ * @return The result of the operation.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_i2c_read_bytes(const dif_i2c_t *i2c, size_t size,
+                                uint8_t *buffer);
```

Comment:
I'd argue that this belongs in an i2c_testutils, not the DIFs. It imposes blocking semantics, where it probably ought to allow for some sort of timeout + bytes actually read, which then might require other IPs for measuring. This is target mode, so our device here doesn't have control of timing.